packages/react-instantsearch-core/src/core/createInstantSearchManager.js

```diff
@@ -319,6 +319,19 @@ export default function createInstantSearchManager({
     }
   }
 
+  function removeDuplicateQuery(params) {
+    let previousIndex = undefined;
+    const queryParamRegex = /&?query=[^&]*/g;
+    return params.replace(queryParamRegex, function replacer(match, offset) {
+      if (previousIndex && offset > previousIndex) {
```

Comment:
```suggestion
      if (typeof previousIndex !== 'undefined' && offset > previousIndex) {
```
Prevents from sneaked zero index (`params = "query=test"`)

Reply:
```suggestion
      if (previousIndex !== undefined && offset > previousIndex) {
```